### PR TITLE
Change default `sync_response_cache_duration`

### LINKED ISSUE #13027: Default `sync_response_cache_duration` can lead to clients never being able to initial sync
### Description

Large accounts or slow servers can mean that the client will retry `/sync` (without `?since`) until it gets a result. However, the server's default cache time for sync responses seemingly went from some number of minutes to zero. This means that the server processes large `/sync` requests in the background but immediately throws away the result if the client gave up waiting and tried a second request.

There should be a non-zero default on the config value.

### Steps to reproduce

- Be in thousands of rooms
- Have a slow-ish server
- Try to sync your account

### Homeserver

t2l.io & t2bot.io

### Synapse Version

1.60.0

### Installation Method

pip (from PyPI)

### Platform

N/A

### Relevant log output

```shell
N/A
```

### Anything else that would be useful to know?

_No response_

### PR DESCRIPTION
**Pull request by erikjohnston**
Fixes #13027.

I picked the value out of thin air, but it'll probably do.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -37,11 +37,8 @@ FROM docker.io/python:${PYTHON_VERSION}-slim as requirements
 # https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/syntax.md#build-mounts-run---mount.
 # Here we use it to set up a cache for apt (and below for pip), to improve
 # rebuild speeds on slow connections.
-RUN \
-   --mount=type=cache,target=/var/cache/apt,sharing=locked \
-   --mount=type=cache,target=/var/lib/apt,sharing=locked \
- apt-get update && apt-get install -y git \
-    && rm -rf /var/lib/apt/lists/*
+RUN apt-get update && apt-get install -y git \
```
Comment (reivilibre) [inline docker/Dockerfile:40]: this seems related

File: docs/usage/configuration/config_documentation.md [hunk 2]
```diff
@@ -1,11 +1,11 @@
 # Configuring Synapse
 
-This is intended as a guide to the Synapse configuration. The behavior of a Synapse instance can be modified 
-through the many configuration settings documented here — each config option is explained, 
+This is intended as a guide to the Synapse configuration. The behavior of a Synapse instance can be modified
+through the many configuration settings documented here — each config option is explained,
 including what the default is, how to change the default and what sort of behaviour the setting governs.
-Also included is an example configuration for each setting. If you don't want to spend a lot of time 
+Also included is an example configuration for each setting. If you don't want to spend a lot of time
```
Comment (reivilibre) [inline docs/usage/configuration/config_documentation.md:6]: the trailing spaces are a bit painful for the diff but I think it's a good idea to have them gone, so let's look over it...

**Review by reivilibre — APPROVED** [review]
LGTM if the Dockerfile changes are removed

File: docker/Dockerfile (hunk 1 above)
Comment (erikjohnston) [reply docker/Dockerfile:40]: SIGH, sorry I did a `git commit -a` accidentally after fixing this up and failed to reject the diff :/

File: docs/usage/configuration/config_documentation.md (hunk 2 above)
Comment (erikjohnston) [reply docs/usage/configuration/config_documentation.md]: Sorry, that wasn't intentional

Comment (erikjohnston) [reply docs/usage/configuration/config_documentation.md]: And have reverted them

**Review by squahtx — APPROVED** [review]
lgtm

**Pull request merged**